### ISSUE TITLE
[Feature request] Add an option to NeedBraces to allow one line statements without braces

### ISSUE THREAD

**Issue by liutikas** [checkstyle/checkstyle#300]
Requesting an option in NeedBraces module to allow if statements without curly braces if they fit on line line.

For example:
```
if (herp.isValid()) return herp;
```

**Comment by liutikas** [comment]
The following code still triggers `'if' construct must use '{}'s.` even with `allowSingleLineIf` set to `true`.
```java
class FooBar {
    FooBar() {
        if (mFoo != null) mFoo.destroy();
    }
}
```

Tested with CheckStyle 6.2

Can we reopen this bug to fix this?

**Comment by liutikas** [comment]
While we are at it can this flag simply allow one line statements for all the `tokens` specified in the module (e.g. include while loops too)

**Comment by romani** [comment]
@liutikas , in that case you might also interested in FOR loop too, am I right ?. What about do-while ?

**Comment by liutikas** [comment]
Yes, all the tokens (for, if, while, ...) with simple one line statements.

**Comment by romani** [comment]
in this case we need an option that will have list of Token where user would like skip violation, to let user define what is ok and what is wrong.

**Comment by liutikas** [comment]
Another option is use the same Token list that exists. Users who want to have a set of Tokens with this skip rule and without it can simply make two modules.

**Comment by romani** [comment]
@liutikas , in that case will skip validation at all, and not only for "one line statement", so it is not complete workaround.
New option is required that will be a list of targets where single line statement is allowed.

**Comment by Mebibyte** [comment]
Any updates on fixing this? Also, the [documentation for this check](http://checkstyle.sourceforge.net/config_blocks.html#NeedBraces) says to set it to FALSE (the default value) to allow single line if statements. Shouldn't it be TRUE?

**Comment by romani** [comment]
@Mebibyte , it for some reason was lost from my tracking, thanks for reminding me.

>  Shouldn't it be TRUE?

No by default we work as before, as more strict.

**Comment by Mebibyte** [comment]
Thanks @romani ! I'm still confused... The default value for allowing single line if statements is false, which means it doesn't allow single line if statements without braces. The example says to allow single line if statements, to set the property to false, which is already the default. Correct?

Either way, the extra property for the check doesn't currently work.

**Comment by romani** [comment]
@Mebibyte, thanks for pointing to mistake in documentation, I put a comment to PR to fix that. 
Default value for allowSingleLineStatement should be FALSE. We have to work as we worked before.
Chekcstyle have to be strict by default and do not allow SingleLineStatements (as single line statements are EVIL and it is hard to debug that statement).

**Comment by alexkravin** [comment]
@Mebibyte, that was a mistake in docs, referenced PR is updating this weirdness.

**Comment by romani** [comment]
fixed, will be release in 6.5

**Comment by liutikas** [comment]
I tested Checkstyle 6.5 and it seems to catch all the correct issues in Chromium code. Thanks!

**Comment by liutikas** [comment]
Setting allowSingleLineStatement to true makes it so checkstyle 6.5 does not complain about:
```java
if (blah) {
  return 1;
} else 
  return 2;
```

It should complain that else statement does not have curly braces.

**Comment by romani** [comment]
@liutikas , do you expect violation on "else" on following code:
```java
if (blah) {
  return 1;
} else return 2;
```

**Comment by liutikas** [comment]
@romani yes I would like that to be a violation.

**Comment by romani** [comment]
@liutikas , can you share your opinion on code examples that are shown in this issue. #1007 (comment)
